src/encoded/commands/deploy.py

```diff
@@ -373,26 +395,58 @@ def _get_run_args(main_args, instances_tag_data):
+        '  # ssh %s' % hostname,
+        '  # %s' % instances_tag_data['id'],
+    ]
+
+
 def _wait_and_tag_instances(main_args, run_args, instances_tag_data, instances, cluster_master=False):
     tmp_name = instances_tag_data['name']
-    domain = 'production' if main_args.profile_name == 'production' else 'instance'
-    for i, instance in enumerate(instances):
-        if main_args.elasticsearch == 'yes' and run_args['count'] > 1:
-            if cluster_master and run_args['master_user_data']:
-                print('Creating Elasticsearch Master Node for cluster')
-                # Hack: current tmp_name was the last data cluster, so remove '4'
-                instances_tag_data['name'] = "{}{}".format(tmp_name[0:-1], 'master')
-            else:
-                print('Creating Elasticsearch cluster')
-                instances_tag_data['name'] = "{}{}".format(tmp_name, i)
+    instances_tag_data['domain'] = 'production' if main_args.profile_name == 'production' else 'instance'
+    output_list = ['']
+    is_cluster_master = False
+    is_cluster = False
+    if main_args.elasticsearch == 'yes' and run_args['count'] > 1:
+        if cluster_master and run_args['master_user_data']:
+            is_cluster_master = True
+            output_list.append('Creating Elasticsearch Master Node for cluster')
         else:
-            instances_tag_data['name'] = tmp_name
+            is_cluster = True
+            output_list.append('Creating Elasticsearch cluster')
+    created_cluster_master = False
+    for i, instance in enumerate(instances):
+        instances_tag_data['name'] = tmp_name
+        instances_tag_data['id'] = instance.id
+        if is_cluster_master:
+            # Hack: current tmp_name was the last data cluster, so remove '4'
+            instances_tag_data['name'] = "{}{}".format(tmp_name[0:-1], 'master')
+        elif is_cluster:
+            instances_tag_data['name'] = "{}{}".format(tmp_name, i)
         if not main_args.spot_instance:
-            print('%s.%s.encodedcc.org' % (instance.id, domain))  # Instance:i-34edd56f
+            if is_cluster_master or (is_cluster and not created_cluster_master):
+                created_cluster_master = True
+                output_list.extend(_get_instance_output(
+                    instances_tag_data,
+                    is_cluster_master=is_cluster_master,
+                ))
+            elif is_cluster:
+                output_list.append('  # %s' % instance.id)
+            elif not is_cluster:
+                output_list.extend(_get_instance_output(instances_tag_data))
             instance.wait_until_exists()
             tag_ec2_instance(instance, instances_tag_data, main_args.elasticsearch, main_args.cluster_name)
-            print('ssh %s.%s.encodedcc.org' % (instances_tag_data['name'], domain))
-            print('https://%s.demo.encodedcc.org' % instances_tag_data['name'])
+    for output in output_list:
+        print(output)
```

Comment:
Not Significant: Do you need additional information along with the print-out..? More information in these of a small explanation.

Reply:
no, all info should be in there already.  Is just collects the prints and outputs once.